app/keepers.go

```diff
@@ -161,17 +161,23 @@ func initStakingKeeper(appCodec codec.Codec, keys map[string]*sdk.KVStoreKey, ke
 	return &stakingK
 }
 
-func initWasmKeeper(appCodec codec.Codec, keys map[string]*sdk.KVStoreKey, keepers *keeperCache, bApp *bam.BaseApp, wasmDir string, wasmConfig wasmtypes.WasmConfig, wasmOpts []wasm.Option) *wasm.Keeper {
+func initWasmKeeper(encodingConfig axelarParams.EncodingConfig, keys map[string]*sdk.KVStoreKey, keepers *keeperCache, bApp *bam.BaseApp, wasmDir string, wasmConfig wasmtypes.WasmConfig, wasmOpts []wasm.Option) *wasm.Keeper {
 	// The last arguments can contain custom message handlers, and custom query handlers,
 	// if we want to allow any custom callbacks
-	wasmOpts = append(wasmOpts, wasmkeeper.WithMessageHandlerDecorator(func(old wasmkeeper.Messenger) wasmkeeper.Messenger {
-		return wasmkeeper.NewMessageHandlerChain(old, nexusKeeper.NewMessenger(getKeeper[nexusKeeper.Keeper](keepers)))
-	}))
+	wasmOpts = append(wasmOpts, wasmkeeper.WithMessageHandlerDecorator(
+		func(old wasmkeeper.Messenger) wasmkeeper.Messenger {
+			encoders := wasm.DefaultEncoders(encodingConfig.Codec, getKeeper[ibctransferkeeper.Keeper](keepers))
```

Comment:
I would also put a comment here to alert people later on. If we need to add any custom encoder for the wasm keeper in the future via `wasmkeeper.WithMessageEncoders`, it would be easy for people to forget to also apply it to this.

Reply:
what would you like to see here? the wasmkeeper comment already states that you should not use `WithMessageHandlerDecorator` together with `WithMessageEncoders`